utils/lib/notebooks/notebook.ex

```diff
@@ -4,22 +4,62 @@ defmodule Utils.Notebooks.Notebook do
   """
   defstruct [
     :content,
-    :index,
     :name,
     :relative_path,
     :title,
     :folder,
-    :type
+    :type,
+    :next_notebook,
+    :prev_notebook
   ]
 
+  @type t :: %__MODULE__{
+          content: String.t(),
+          name: String.t(),
+          relative_path: String.t(),
+          title: String.t(),
+          folder: String.t(),
+          type: String.t(),
+          next_notebook: map(),
+          prev_notebook: map()
+        }
+
+  require Logger
+
+  @doc """
+  Given a list of names and relative paths will create a list of notebook structs
+  with the next_notebook and prev_notebook attributes set according to the position 
+  of the notebook in the list.
+
+  Useful for parsing input that is read from the start.livemd file.
+  """
+  @spec map_notebooks([%{name: String.t(), relative_path: String.t()}]) :: [__MODULE__.t()]
+  def map_notebooks(nodes) do
```

Comment:
What about instead of storing the next and the prev in the notebooks we create chunks for the prev/current/next trios? Here's an example that handles the beginning and end edge cases.

```elixir
[first, second | _tail] = notebooks 
[[first, second] | Enum.chunk_every(notebooks , 3, 1)]
```